transforms/manual-bind-to-arrow.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+/**
+ * class Component extends React.Component {
+ *   constructor() { this.onClick = this.onClick.bind(this); }
+ *   onClick() { }
+ * }
+ *
+ * -->
+ *
+ * class Component extends React.Component {
+ *   constructor() { }
+ *   onClick = () => { }
+ * }
+ */
+
+export default function transformer(file, api) {
+  const j = api.jscodeshift;
+  const {expression, statement, statements} = j.template;
+
+  var root = j(file.source);
+
+  return root
+    .find(j.AssignmentExpression)
+    .filter(path => {
+      // Check constructor
+      var methodPath = path;
+      while (methodPath &&
+             (methodPath.node.type !== 'MethodDefinition' ||
+              methodPath.node.kind !== 'constructor')) {
+        methodPath = methodPath.parentPath;
+      }
+      if (!methodPath) {
+        return false;
+      }
+
+      if (!(
+        path.node.left.type === 'MemberExpression' &&
+        path.node.left.object.type === 'ThisExpression' &&
+        path.node.left.property.type === 'Identifier' &&
+        path.node.right.type === 'CallExpression' &&
+        path.node.right.callee.type === 'MemberExpression' &&
+        path.node.right.callee.property.type === 'Identifier' &&
+        path.node.right.callee.property.name === 'bind' &&
+        path.node.right.callee.object.type === 'MemberExpression' &&
+        path.node.right.callee.object.property.type === 'Identifier' &&
+        path.node.right.callee.object.object.type === 'ThisExpression' &&
+        path.node.left.property.name === path.node.right.callee.object.property.name
+      )) {
+        return false;
+      }
+
```

Comment:
Thanks for the trick! In the new version I've added conditionals inside of that logic so I'm not really sure we can do it as you mentioned.

Also, for the method lookup, I need the reference to that node, so I'd need to traverse it up anyway.